documentation/src/main/asciidoc/topics/ref_eviction_examples.adoc

```diff
@@ -1,92 +1,92 @@
 [id='eviction_example-{context}']
 = Eviction Examples
 
-You configure eviction in `infinispan.xml` as part of your cache definition.
+You configure eviction as part of your cache definition.
 
 .Default memory configuration
 
 Eviction is not enabled, which is the default configuration. {brandname} stores
-cache entries as objects in the data container.
+cache entries as objects in the JVM heap.
 
 [source,xml,options="nowrap",subs=attributes+]
 ----
 <memory />
 ----
 
-.Passivation with eviction
+.Eviction based on the total number of entries
 
-Passivation persists data to cache stores when {brandname} evicts entries. You
-should always enable eviction if you enable passivation.
+{brandname} stores cache entries as objects in the JVM heap. Eviction happens
+when there are 100 entries in the data container and {brandname} gets a request
+to create a new entry:
 
 [source,xml,options="nowrap",subs=attributes+]
 ----
-<persistence passivation="true">
-  ...
-</persistence>
-
-<memory />
+<memory max-count="100"/>
 ----
 
-.Manual eviction
+.Eviction based maximum size in bytes
+
+{brandname} can store cache entries as `byte[]` arrays if you encode entries in binary storage formats, for example: `application/x-protostream` format.
 
-{brandname} stores cache entries as objects. Eviction is not enabled but
-performed manually using the `evict()` method.
+In the following example, {brandname} performs eviction when the size of the
+data container reaches 500 MB (megabytes) in size and it gets a request to
+create a new entry:
 
 [source,xml,options="nowrap",subs=attributes+]
 ----
-<memory>
-  <object strategy="MANUAL" />
-</memory>
+<encoding media-type="application/x-protostream"/> <1>
+<memory max-size="500 MB"/> <2>
 ----
 
-.Object storage with eviction
+<1> Specifies a binary format for entries in the cache.
+<2> Defines the maximum size of the data container as MB (megabytes).
 
-{brandname} stores cache entries as objects. Eviction happens when there are
-100 entries in the data container and {brandname} gets a request to create a
-new entry:
+.Off-heap storage
+
+{brandname} stores cache entries as bytes in native memory. Eviction happens
+when there are 100 entries in the data container and {brandname} gets a request
+to create a new entry:
 
 [source,xml,options="nowrap",subs=attributes+]
 ----
-<memory>
-  <object size="100" />
-</memory>
+<memory storage="OFF_HEAP" max-count="100"/>
```

Comment:
Believe we should have the media-type here since OFF_HEAP doesn't work with object media type.

Reply:
@wburns I took this from the example in the upgrade guide here: https://github.com/infinispan/infinispan/blob/master/documentation/src/main/asciidoc/topics/upgrading.adoc 

maybe I'm not clear on some of the details. I get that off heap is always represented as byte[] so the object media type is not suitable. but should users always configure the encoding when using off heap?

Reply:
@gustavonalle we should update the upgrading guide as well it seems too then and more clarification would be great.

Reply:
When using OFF_HEAP it automatically uses the "user marshaller"

Reply:
@wburns and @gustavonalle thanks, I've added this to the configuring Off Heap topic. here's the readable HTML: http://file.rdu.redhat.com/dnaro/ISPN-11869/titles/configuring/configuring.html#configure_off_heap-config